android/src/main/java/com/rnbiometrics/ReactNativeBiometrics.java

```diff
@@ -144,6 +179,99 @@ public void deleteKeys(Promise promise) {
         }
     }
 
+    @ReactMethod
+    public void deleteEncryptionKey(Promise promise) {
+        boolean deletionSuccessful = false;
+        if (doesBiometricEncryptionKeyExist()) {
+            deletionSuccessful = deleteBiometricEncryptionKey();
+            if (!deletionSuccessful) {
+                promise.reject("Error deleting biometric encryption key", "Error deleting biometric encryption key");
+            }
+        }
+        WritableMap resultMap = new WritableNativeMap();
+        resultMap.putBoolean("keysDeleted", deletionSuccessful);
+        promise.resolve(resultMap);
+    }
+
+    @ReactMethod
+    public void encryptData(final ReadableMap params, final Promise promise) {
+        UiThreadUtil.runOnUiThread(
+                new Runnable() {
+                    @Override
+                    public void run() {
+                        try {
+                            if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
+                                throw new Exception("Cannot generate keys on android versions below 6.0");
+                            }
+                            Cipher cipher = Cipher.getInstance("AES/GCM/NoPadding");
+                            KeyStore keyStore = KeyStore.getInstance("AndroidKeyStore");
+                            keyStore.load(null);
+
+                            cipher.init(Cipher.ENCRYPT_MODE, (SecretKey) keyStore.getKey(biometricEncryptionKeyAlias, null));
+
+                            BiometricPrompt biometricPrompt = new BiometricPrompt(
+                                    (FragmentActivity) getCurrentActivity(),
+                                    Executors.newSingleThreadExecutor(),
+                                    new EncryptDataCallback(promise, params.getString("payload"))
+                            );
+
+                            PromptInfo promptInfo = new PromptInfo.Builder()
+                                    .setDeviceCredentialAllowed(false)
```

Comment:
Allowing device credentials is a new aspect being added into this library. I would like to either implement that capability or note in the documentation that it is not supported.

Reply:
Yes, the customer who contracted us for the feature needed it paired with strong biometrics, so we didn't bother with a fallback path, esp. since all the design work was done pre Android 11. (And 11 did have a bunch of bugs in it too, iirc. It's been a while.)

I think these days it'd be more reasonable to support them? 

@f-23 Could you look into supporting `allowDeviceCredentials` here?